EKF/control.cpp

```diff
@@ -789,21 +789,17 @@ void Ekf::controlHeightSensorTimeouts()
 
 		bool request_height_reset = false;
 
-		// handle the case where we are using baro for height
 		if (_control_status.flags.baro_hgt) {
 			// check if GPS height is available
 			const gpsSample &gps_init = _gps_buffer.get_newest();
 			const bool gps_hgt_accurate = (gps_init.vacc < _params.req_vacc);
 
-			const baroSample &baro_init = _baro_buffer.get_newest();
-			const bool baro_data_available = isRecent(baro_init.time_us, 2 * BARO_MAX_INTERVAL);
-
-			// check for inertial sensing errors in the last 10 seconds
+			// check for inertial sensing errors in the last BADACC_PROBATION seconds
 			const bool prev_bad_vert_accel = isRecent(_time_bad_vert_accel, BADACC_PROBATION);
 
 			// reset to GPS if adequate GPS data is available and the timeout cannot be blamed on IMU data
 			const bool reset_to_gps = !_gps_hgt_intermittent &&
-					    ((gps_hgt_accurate && !prev_bad_vert_accel) || !baro_data_available);
+					    ((gps_hgt_accurate && !prev_bad_vert_accel) || _baro_hgt_faulty);
```

Comment:
`_baro_hgt_faulty` will not only be set by the isRecent check. Are you aware of that?

Reply:
Yes, the 2nd place where it is set is 3 lines below